doc/man7/passphrase-encoding.pod

```diff
@@ -0,0 +1,182 @@
+The standard stipulates that the pass phrase shall be encoded as an ASN.1
+BMPString, which consists of the code points of the basic multilingual plane,
+encoded in big endian (UCS-2 BE).
+
+OpenSSL tries to adapt to this requirements in one of the following manners:
+
+=over 4
+
+=item 1.
+
+Treats the received pass phrase as UTF-8 encoded and tries to re-encode it to
+UTF-16 (which is the same as UCS-2 for characters U+0000 to U+D7FF and U+E000
+to U+FFFF, but becomes an expansion for any other character), or failing that,
+proceeds with step 2.
+
+=item 2.
+
+Assumes that the pass phrase is encoded in ASCII or ISO-8859-1 and
+opportunistically prepends each byte with a zero byte to obtain the UCS-2
+encoding of the characters, which it stores as a BMPString.
+
+Note that since there is no check of your locale, this may produce UCS-2 /
+UTF-16 characters that do not correspond to the original pass phrase characters
+for other character sets, such as any ISO-8859-X encoding other than
+ISO-8859-1.
+
+=back
+
+OpenSSL versions older than 1.1.0 do variant 2 only, and that is the reason why
+OpenSSL still does this, to be able to read files produced with older versions.
+
+It should be noted that this approach isn't entirely fault free.
+
+A passphrase encoded in ISO-8859-2 could very well have a sequence such as
+0xC3 0xAF (which is the two characters "LATIN CAPITAL LETTER A WITH BREVE"
+and "LATIN CAPITAL LETTER Z WITH DOT ABOVE" in ISO-8859-2 encoding), but would
+be misinterpreted as the perfectly valid UTF-8 encoded code point U+00EF (LATIN
+SMALL LETTER I WITH DIARESIS) I<if the passphrase doesn't contain anything that
+would be invalid UTF-8>.
+A pass phrase that contains this kind of byte sequence will give a different
+outcome in OpenSSL 1.1.0 and newer than in OpenSSL older than 1.1.0.
+
+ 0x00 0xC3 0x00 0xAF                    # OpenSSL older than 1.1.0
+ 0x00 0xEF                              # OpenSSL 1.1.0 and newer
+
+On the same accord, anything encoded in UTF-8 that was given to OpenSSL older
+than 1.1.0 was misinterpreted as ISO-8859-1 sequences.
+
+=head2 OSSL_STORE
+
+L<ossl_store(7)> acts as a general interface to access all kinds of objects,
+potentially protected with a pass phrase, a PIN or something else.
+This API currently doesn't stipulate any specific encoding of pass phrases, but
+uses the underlying routines with their behaviours.
+This means that when using the built-in C<file:> scheme loader, the pass phrase
+to unlock a PKCS#12 file will be treated as described for PKCS#12 above, and
+the pass phrase for a PEM files will be treated as the general case described
+above, since that loader uses the same underlying routines.
+I<Note that other loaders will have their own behaviours>.
+
+=head1 RECOMMENDATIONS
+
+This section assumes that you know what pass phrase you should type in is, but
+that it may have been used in a different character encoding than your current
+environment gives you by default.
```

Comment:
> I'm not an expert in English, but I find notion of "environment giving me encoding" somewhat strange. 

I agree with @dot-asm. How about the following alternative formulation, which uses@dot-asm's suggestion above?

> This section assumes that you know what pass phrase was used for encryption, but that it may have been encoded by the application in a different character encoding than the one used by your application's current input method.

Reply:
Is "input method" a term that's known well enough?  *I* understand all too well what it means, but...

@mspncp, my first question when reading that sentence is "what application are you talking about"?  It's most certainly not `openssl`!

Reply:
> This section assumes that you know what pass phrase was used for encryption, but that it may have been encoded ~~by the application~~ in a different character encoding than the one used by your ~~application's~~ current input method.

Better?

Reply:
So you seem to agree that "input method" is a well enough known term.  Then sure, that sentence seem fine.

Reply:
Let’s say it that way: I don’t have a better alternative.